scripts/generate-icon-react-wrapper.js

```diff
@@ -0,0 +1,172 @@
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import { createComponent } from '@lit-labs/react';
+import * as React from 'react';
+  
+import { ${component} as ${componentAliasName} } from '@spectrum-web-components/${iconPkg}/src/elements/${id}.js';
+import '@spectrum-web-components/${iconPkg}/icons/${iconElementName}.js';
+  
+export const ${component} = createComponent({ react: React, tagName: '${iconElementName}', elementClass: ${componentAliasName}, events: {}, displayName: '${component}' });
+
+export type ${component}Type = ${componentAliasName};
+`;
+};
+
+/**
+ * Core entry function
+ */
+const generateIconWrapper = async (iconType) => {
+    glob(
+        path.resolve(
+            __dirname,
+            '..',
+            `packages/${iconType}/src/elements/**.d.ts`
+        ),
+        async (_, icons) => {
+            for (let icon of icons) {
+                const id = path
+                    .basename(icon)
+                    .split('.')[0]
+                    .substring('Icon'.length);
+                const componentName =
+                    id === 'github' ? 'GitHub' : Case.pascal(id);
+                const iconElementName = `sp-icon-${Case.kebab(componentName)}`;
+                await outputFile(
+                    path.resolve(
+                        __dirname,
+                        '..',
+                        `react/${iconType}/${componentName}.ts`
+                    ),
+                    prettier.format(
+                        index(
+                            `Icon${componentName}`,
+                            `Icon${id}`,
+                            iconElementName,
+                            `${iconType}`
+                        ),
+                        {
+                            parser: 'babel',
+                            ...prettierConfig,
+                        }
+                    )
+                );
+            }
+
+            const { name: pkgName, version: pkgVersion } = await readJSON(
+                path.resolve(
+                    __dirname,
+                    '..',
+                    `packages/${iconType}/package.json`
+                )
+            );
+
+            await outputFile(
+                path.resolve(__dirname, '..', `react/${iconType}/package.json`),
+                prettier.format(
+                    `{
+    "name": "@swc-react/${iconType}",
+    "version": "${pkgVersion}",
+    "publishConfig": {
+        "access": "public"
+    },
+    "description": "React wrapper of the ${iconType} component",
+    "license": "Apache-2.0",
+    "author": "",
+    "main": "index.js",
+    "files": [
+        "**/*.d.ts",
+        "**/*.js",
+        "**/*.js.map"
+    ],
+    "keywords": [
+        "React",
+        "Spectrum Web Components",
+        "${iconType}"
+    ],
+    "dependencies": {
+        "@lit-labs/react": "1.1.1",
```

Comment:
@jianliao getting to the end here and taking a quick last look.

Do you see any issue with opening this dependency to `^1.1.1` so that consumers can take newer versions?

Reply:
Just realizing that this is true in the non-icon version. To ensure we keep these two approaches aligned, does it make sense to abstract a couple pieces of these two wrapper files to a shared set of utilities?

Reply:
I've moved all the code logic to cem file to improve the code reuse and code gen template reuse.